packages/admin/cms-admin/src/pages/pageTree/useExtractPages.tsx

```diff
@@ -0,0 +1,297 @@
+                        throw new Error(`Error while fetching page`);
+                    }
+                }),
+            );
+        },
+        [client, documentTypes, locale],
+    );
+
+    const getContentsFromClipboard = async (): Promise<GetContentsFromClipboardResponse> => {
+        const text = await readClipboard();
+
+        if (text === undefined) {
+            return {
+                canPaste: false,
+                error: (
+                    <FormattedMessage
+                        id="comet.pages.cannotPastePage.messageFailedToReadClipboard"
+                        defaultMessage="Can't read clipboard content. Please make sure that clipboard access is given"
+                    />
+                ),
+            };
+        }
+
+        if (text.trim() === "") {
+            return {
+                canPaste: false,
+                error: <FormattedMessage id="comet.pages.cannotPastePage.messageEmptyClipboard" defaultMessage="Clipboard is empty" />,
+            };
+        }
+
+        try {
+            const parsedText = getCSVAsContent(text);
+
+            if (isContentsClipboard(parsedText)) {
+                return {
+                    canPaste: true,
+                    content: {
+                        contents: Object.entries(parsedText.textContents).map((content) => ({ original: content[0], replaceWith: content[1] })),
+                    },
+                };
+            } else {
+                return {
+                    canPaste: false,
+                    error: (
+                        <FormattedMessage
+                            id="comet.pages.cannotPasteContent.messageFailedToParseClipboard"
+                            defaultMessage="Content from clipboard isn't valid document/block content"
+                        />
+                    ),
+                };
+            }
+        } catch (e) {
+            return {
+                canPaste: false,
+                error: (
+                    <FormattedMessage
+                        id="comet.pages.cannotPasteContent.messageFailedToParseClipboard"
+                        defaultMessage="Content from clipboard isn't valid document/block content"
+                    />
+                ),
+            };
+        }
+    };
+
+    const getContentAsCSV = (contents: string[]) => {
+        const rows = ["Original;ReplaceWith"];
+
+        contents.forEach((content) => {
+            const value = content.replace(/;/g, '"";""');
+            rows.push(`${value};${value}`);
+        });
+        return rows.join("\n");
+    };
+
+    const getCSVAsContent = (csv: string) => {
+        const content: { [key: string]: string } = {};
+
+        const lines = csv.split("\n");
+
+        for (let i = 1; i < lines.length; i++) {
+            const lineContent = lines[i].split(";");
+
+            content[lineContent[0]] = lineContent[1];
+        }
+
+        return { textContents: content };
+    };
```

Comment:
Both functions don't rely on anything related to React state etc. and could be moved outside the hook. I'd also improve the naming: `convertTextContentsToCsv` and `convertCsvToTextContents`